### ISSUE TITLE
Best pipeline trained by AutoMLSearch gets different score than cloned version trained on `X_train`

### ISSUE THREAD

**Issue by angela97lin** [alteryx/evalml#2844]

**Comment by freddyaboulton** [comment]
@angela97lin Does it happen for regression problems? Could it be threshold tuning? `AutoMLSearch`'s `train_best_pipeline` will tune the threshold by default I believe.

**Comment by angela97lin** [comment]
@freddyaboulton Nope, it doesn't happen for regression problems so I think you're correct. Setting `optimize_thresholds` to False (defaults to True) also changes the behavior s.t. the two results are the same.

Next question then is: Do we want this type of behavior / is this intentional? I can see why we would want this, but question if it's a little confusing for users. Only the best pipeline is able to trained with the threshold, right? 

1) Is it confusing that a "cloned" version of the pipeline external to automl search would not produce the same results? How do we make this more clear?
2) If a user wanted to choose pipeline with id=0, there's not a good way to do so without digging into automl search and the tuned threshold parameter in the same way that we do for the best pipeline, right? I think it can be confusing / unclear that grabbing the best pipeline will be fitted using the threshold, but grabbing any other pipeline and manually training it will not use the fitted threshold. 

TLDR: the behavior seems a bit inconsistent and can be confusing or lead to misunderstandings. Is this what we're okay with, or should we clarify this somehow?

**Comment by freddyaboulton** [comment]
@angela97lin Great points. I think this behavior is intentional. I think it was introduced in these PRs:

https://github.com/alteryx/evalml/pull/1943 Made it so that we tuned thresholds by default
https://github.com/alteryx/evalml/pull/2320 Added a secondary objective that we use for tuning when the primary objective is not tunable. 

> Only the best pipeline is able to trained with the threshold, right?

Every pipeline during automl search is trained with threshold tuning because of the default values set in #1943 and #2320. 

So the problem is that AutoMLSearch is tuning thresholds by default but there's no way for users to know that/recreate that outside of search if they wanted to "export" the pipelines out of AutoMLSearch.

Would one of the two fix the problem:
1. Store the optimal threshold value for each pipeline in `AutoMLSearch.results` and make sure `get_pipeline` sets the threshold value correctly?
2. Make threshold optimization a step in the pipeline?

**Comment by dsherry** [comment]
Discussion looks good. Whoever picks this up, we need to decide on what to do next!

Questions from refinement:
* Does cloning copy the bin class threshold? (We in refinement think no)
* If not, why not?
